### PR TITLE
Integration tests to check the status of the access tokens of a user after disabling the user account

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
This integration test will check whether a user's access tokens generated for multiple service provider applications has been revoked after disabling the user account.

Addressed Issue: https://github.com/wso2/product-is/issues/14048

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 1]
```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
 		return appDtoResult;
 	}
 
+	/**
+	 * Create Application with a given appDTO and service provider name
```
Comment (rajithacharith) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:214]: End comment lines with a full stop (`.`).

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 2]
```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
...
+
+		serviceProvider = appMgtclient.getApplication(serviceProviderName);
+		serviceProvider = setServiceProviderClaimConfig(serviceProvider);
+		serviceProvider.setOutboundProvisioningConfig(new OutboundProvisioningConfig());
+		List<InboundAuthenticationRequestConfig> authRequestList = new ArrayList<>();
+
+		if (consumerKey != null) {
+			InboundAuthenticationRequestConfig opicAuthenticationRequest = new InboundAuthenticationRequestConfig();
+			opicAuthenticationRequest.setInboundAuthKey(consumerKey);
+			opicAuthenticationRequest.setInboundAuthType("oauth2");
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:244]: I am guess you meant OIDC here?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 3]
```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
...
 	}
 
+	/**
+	 * Create Application with a given appDTO and service provider name
+	 *
+	 * @return OAuthConsumerAppDTO
+	 * @throws Exception
+	 */
+	public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO, String serviceProviderName) throws Exception {
+		OAuthConsumerAppDTO appDtoResult = null;
```
Comment (rajithacharith) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:220]: eave a blank line after method signature

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 4]
```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
...
+			authRequestList.add(opicAuthenticationRequest);
+		}
+
+		String openidRealm = serviceProviderName;
+		if (openidRealm != null) {
+			InboundAuthenticationRequestConfig opicAuthenticationRequest = new InboundAuthenticationRequestConfig();
+			opicAuthenticationRequest.setInboundAuthKey(openidRealm);
+			opicAuthenticationRequest.setInboundAuthType("openid");
+			authRequestList.add(opicAuthenticationRequest);
+		}
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:269]: If you are testing oauth or OIDC you don't need to set these as they are different protocols which your app wouldn't use

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 5]
```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
...
+		OAuthConsumerAppDTO appDtoResult = null;
+
+		adminClient.registerOAuthApplicationData(appDTO);
+		OAuthConsumerAppDTO[] appDtos = adminClient.getAllOAuthApplicationData();
+
+		for (OAuthConsumerAppDTO appDto : appDtos) {
+			appDtoResult = appDto;
+			consumerKey = appDto.getOauthConsumerKey();
+			consumerSecret = appDto.getOauthConsumerSecret();
+		}
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:229]: What are we trying to do with this code segment?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 1 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 2 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 3 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 4 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 5 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 6]
```diff
@@ -148,26 +148,27 @@ ServiceProvider setServiceProviderClaimConfig(ServiceProvider serviceProvider) {
...
+	public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO, String serviceProviderName)
+			throws Exception {
 		OAuthConsumerAppDTO appDtoResult = null;
 
 		adminClient.registerOAuthApplicationData(appDTO);
 		OAuthConsumerAppDTO[] appDtos = adminClient.getAllOAuthApplicationData();
 
 		for (OAuthConsumerAppDTO appDto : appDtos) {
-			if (appDto.getApplicationName().equals(OAuth2Constant.OAUTH_APPLICATION_NAME)) {
+			if (appDto.getApplicationName().equals(appDto.getApplicationName())) {
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java:159]: what are we trying to do here?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+        String tenantCookie = logManager.login(ADMIN + "@" + secondaryTenantDomain,
+                ADMIN, isServer.getInstance().getHosts().get("default"));
+        tenantIDPMgtClient = new IdentityProviderMgtServiceClient(tenantCookie, backendURL);
+    }
+
+    @Test(description = "Create access tokens")
+    public void testCreateAccessTokens() throws Exception {
+
+        Set<String> appKeys = applications.keySet();
+        for (String app : appKeys) {
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:172]: app->appName

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 6 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Trying to validate the creation of OAuth application. Fixed typo issue on https://github.com/wso2/product-is/pull/14727/commits/4d3f5370d709c1e5df86fce3f4082ba2683b31d5

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java (hunk 7 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/4d3f5370d709c1e5df86fce3f4082ba2683b31d5

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+    private static final String TENANT_DOMAIN = "wso2.com";
+
+    private static final String PROFILE_NAME = "default";
+    private static final String TEST_USER_USERNAME = "testUser";
+    private static final String TEST_USER_PASSWORD = "Ab@123";
+    private static final String TEST_ROLE = "testRole";
+    private static final String ADMIN = "admin";
+
+    private static final String ACCOUNT_DISABLED_CLAIM_URI = "http://wso2.org/claims/identity/accountDisabled";
+    private static final String ENABLE_ACCOUNT_DISABLING_PROPERTY = "account.disable.handler.enable";
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:108]: Check for existing constants

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+                            privilegedAccessTokens.get(appName));
+            Assert.assertTrue(revokedTokenIntrospectionResponse.indicatesSuccess(),
+                    "Failed to receive a success response.");
+            Assert.assertFalse(revokedTokenIntrospectionResponse.toSuccessResponse().isActive(),
+                    "Introspection response of a revoked access token is successful.");
+        }
+    }
+
+    private void createServiceProviderApplication() throws Exception {
+
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:234]: Edit code snippet duplication

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 10]
```diff
@@ -0,0 +1,445 @@
...
+        applications.put(SERVICE_PROVIDER_2_NAME, oAuthConsumerAppDTO2);
+    }
+
+    private void addNewTestUserWithRole() throws Exception {
+
+        remoteUSMServiceClient.addUser(TEST_USER_USERNAME, TEST_USER_PASSWORD, null, null,
+                PROFILE_NAME, false);
+        userMgtClient = new UserManagementClient(backendURL, sessionCookie);
+        userMgtClient.addRole(TEST_ROLE, new String[] {TEST_USER_USERNAME},
+                new String[] {"/permission/admin/login"}, false);
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:262]: New role is not mandatory. Double check!

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 11]
```diff
@@ -0,0 +1,445 @@
...
+        disableAccountClaim.setFieldValue(claimValue);
+        UserFieldDTO[] fields = new UserFieldDTO[1];
+        fields[0] = disableAccountClaim;
+        profile.setFieldValues(fields);
+        userProfileMgtClient.setUserProfile(TEST_USER_USERNAME, profile);
+    }
+
+    private void updateResidentIDP(IdentityProvider residentIdentityProvider, boolean isSuperTenant) throws Exception {
+
+        FederatedAuthenticatorConfig[] federatedAuthenticatorConfigs =
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:281]: Not required

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 12]
```diff
@@ -0,0 +1,445 @@
...
+                        ArrayUtils.removeElement(federatedAuthenticatorConfigs,
+                                authenticatorConfig);
+            }
+        }
+        residentIdentityProvider.setFederatedAuthenticatorConfigs(federatedAuthenticatorConfigs);
+        if (isSuperTenant) {
+            idPMgtClient.updateResidentIdP(residentIdentityProvider);
+        } else {
+            tenantIDPMgtClient.updateResidentIdP(residentIdentityProvider);
+        }
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:295]: Check for an updateResidentIdP method

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 13]
```diff
@@ -0,0 +1,445 @@
...
+        String accountDisabledClaimValue = null;
+        if (ArrayUtils.isNotEmpty(claimValues)) {
+            accountDisabledClaimValue = claimValues[0].getValue();
+        }
+        Assert.assertTrue(Boolean.parseBoolean(accountDisabledClaimValue), "User account didn't disabled");
+    }
+
+    @Test(description = "Check whether access token is revoked after disabling the account",
+            dependsOnMethods = "disableUserAccount")
+    private void introspectAccessTokenOfDisabledAccount() throws Exception {
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:219]: End the method name with "test"

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java (hunk 9 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:234]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/4882f9bbdb620ceaaf5e04a15be585a6c1676420

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java (hunk 10 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/4882f9bbdb620ceaaf5e04a15be585a6c1676420

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java (hunk 13 above)
Comment (ZiyamSanthosh) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java]: Fixed in https://github.com/wso2/product-is/pull/14727/commits/4882f9bbdb620ceaaf5e04a15be585a6c1676420

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java [hunk 14]
```diff
@@ -0,0 +1,432 @@
...
+    private UserIdentityManagementAdminServiceClient userIdentityManagementAdminServiceClient;
+    private ApplicationManagementServiceClient applicationManagementServiceClient;
+    private RemoteUserStoreManagerServiceClient usmClient;
+
+    private final String tokenType;
+    private final String adminUsername;
+    private final String adminPassword;
+    private final String activeTenant;
+    private static final String TENANT_DOMAIN = "wso2.com";
+
```
Comment (ZiyamSanthosh) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java:100]: Check the gaps

**Review by inthirakumaaran — APPROVED** [review]

**Comment by inthirakumaaran** [comment]
Run the integration tests before merging the PR

**Pull request merged**